pkg/server/plugin/nodeattestor/awsiid/eks_test.go

```diff
@@ -0,0 +1,323 @@
+	testEKSValidator := buildEKSValidationClient()
+	testEKSClient := newFakeEKSClient()
+	testASGClient := newFakeASGClient()
+
+	// pass valid node
+	ok, err := testEKSValidator.IsNodeInCluster(context.Background(), testEKSClient, testASGClient, testEKSInstanceID)
+	require.NoError(t, err)
+	require.Equal(t, true, ok)
+
+	// fail when node doesn't exist
+	ok, err = testEKSValidator.IsNodeInCluster(context.Background(), testEKSClient, testASGClient, "i-nonexistent")
+	require.NoError(t, err)
+	require.Equal(t, false, ok)
+}
+
+func TestCheckIfEKSNodeListIsStale(t *testing.T) {
+	testEKSValidator := buildEKSValidationClient()
+
+	testIsStale := testEKSValidator.checkIfEKSNodeListIsStale()
+	require.True(t, testIsStale)
+
+	_, err := testEKSValidator.reloadNodeList(context.Background(), newFakeEKSClient(), newFakeASGClient(), false)
+	require.NoError(t, err)
+	testIsStale = testEKSValidator.checkIfEKSNodeListIsStale()
+	require.False(t, testIsStale)
+}
+
+func TestReloadNodeList(t *testing.T) {
+	testEKSValidator := buildEKSValidationClient()
+	testEKSClient := newFakeEKSClient()
+	testASGClient := newFakeASGClient()
+
+	// check once config is provided correctly and catchburst is false, node list populated along with timestamp
+	_, err := testEKSValidator.reloadNodeList(context.Background(), testEKSClient, testASGClient, false)
+	require.NoError(t, err)
+	require.Len(t, testEKSValidator.eksNodeList, 2) // Two instances in the test setup
+	require.Greater(t, testEKSValidator.eksNodeListValidDuration, time.Now())
+	require.Equal(t, testEKSValidator.retries, eksRetries)
+
+	// check if the list of nodes is updated when catchburst is true
+	// but the timestamp is not updated
+	existingValidDuration := testEKSValidator.eksNodeListValidDuration
+	testEKSValidator.eksNodeList = make(map[string]any)
+	_, err = testEKSValidator.reloadNodeList(context.Background(), testEKSClient, testASGClient, true)
+	require.NoError(t, err)
+	require.Equal(t, existingValidDuration, testEKSValidator.eksNodeListValidDuration)
+	require.Len(t, testEKSValidator.eksNodeList, 2)
+
+	// set retry to 0 and make sure the list is not updated
+	testEKSValidator.retries = 0
+	testEKSValidator.eksNodeList = make(map[string]any)
+	_, err = testEKSValidator.reloadNodeList(context.Background(), testEKSClient, testASGClient, true)
+	require.NoError(t, err)
+	require.Empty(t, testEKSValidator.eksNodeList)
+
+	// make sure retry is reset, once we are over TTL
+	// move clock ahead by 1 minute. And as our TTL is 30 seconds, it should refresh the list
+	testEKSValidator = buildEKSValidationClient()
+	_, err = testEKSValidator.reloadNodeList(context.Background(), testEKSClient, testASGClient, false)
+	require.NoError(t, err)
+	require.Len(t, testEKSValidator.eksNodeList, 2)
+	testEKSValidator.clk = buildEKSNewMockClock(1*time.Minute, testEKSClockMutAfter)
+	testEKSValidator.retries = 0 // trigger refresh to reset retries
+	require.Equal(t, testEKSValidator.retries, 0)
```

Comment:
This assertion feels unnecessary to me since we are not running multiple goroutines in this test